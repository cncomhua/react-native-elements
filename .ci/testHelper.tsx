import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider, FullTheme, colors } from '../src/config';
import deepmerge from 'deepmerge';
import { View } from 'react-native';

export const renderWithTheme = (
  children: React.ReactChild,
  themeProp: Partial<FullTheme> = {},
  options?: RenderOptions
) => {
  return render(
    <ThemeProvider theme={deepmerge({ colors }, themeProp)}>
      {children}
    </ThemeProvider>,
    options
  );
};

// used for getting findByType e.g. wrapper.findByType(View)
export const wrapperFindByType = (children: any) => {
  const { queryByTestId } = renderWithTheme(
    <View testID="wrapper">{children}</View>
  );
  const wrapper = queryByTestId('wrapper');
  return wrapper;
};
