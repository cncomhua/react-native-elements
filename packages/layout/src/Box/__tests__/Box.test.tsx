import React from 'react';
import { renderWithWrapper } from '../../../.ci/testHelper';
import { Box } from '..';

describe('Button Component', () => {
  it('should match snapshot', () => {
    const { toJSON } = renderWithWrapper(<Box m={2} />, 'RNE_BUTTON_WRAPPER');
    expect(toJSON()).toMatchSnapshot();
  });
});
