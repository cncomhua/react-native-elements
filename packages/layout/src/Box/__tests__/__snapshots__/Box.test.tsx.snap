// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": undefined,
      "height": undefined,
      "margin": 2,
      "marginBottom": undefined,
      "marginHorizontal": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "marginVertical": undefined,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "paddingVertical": undefined,
      "width": undefined,
    }
  }
/>
`;
